### PR TITLE
adds purely informational alerts to the fault bus

### PR DESCRIPTION
**Pull request by ryanfkeepers**
Adds a new type of entry to the fault bus: Alerts. These values are for non-erroneous messages that we want 1/ clearly displayed to the user at the end of an operation and 2/ persisted with the backup for later review.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #4264

#### Test Plan

- [x] :zap: Unit test

### CODE REVIEW

File: src/pkg/fault/fault.go [hunk 1]
```diff
@@ -194,6 +205,35 @@ func (e *Bus) addSkip(s *Skipped) *Bus {
...
+// Importantly, alerts are not errors, exceptions, or skipped items.
+// An alert should only be generated if no other fault functionality
+// is in use, but that we still want the end user to clearly and
+// plainly receive a notification about a runtime event.
+func (e *Bus) AddAlert(ctx context.Context, a *Alert) {
+	if a == nil {
+		return
+	}
+
+	e.mu.Lock()
```
Comment (meain) [inline src/pkg/fault/fault.go:218]: Should we move the locking to `addAlert`?

**Review by meain — APPROVED** [review]
LGTM

**Review by pandeyabs — APPROVED** [review]
Looks great! Left a few nits.

File: src/pkg/fault/item.go [hunk 2]
```diff
@@ -264,3 +264,67 @@ func itemSkip(t itemType, cause skipCause, namespace, id, name string, addtl map
...
+}
+
+func (a Alert) MinimumPrintable() any {
+	return a
+}
+
+// Headers returns the human-readable names of properties of a skipped Item
+// for printing out to a terminal.
+func (a Alert) Headers() []string {
+	return []string{"Action", "Message", "Container", "Name", "ID"}
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/pkg/fault/item.go:302]: Fair.  Will probably address in a separate PR.

File: src/pkg/fault/fault.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/pkg/fault/fault.go:218]: Hmm, will follow up in a different PR.  This has some thorny side effects that I need to even out.

**Pull request merged**